driver/utils/type_info.h

```diff
@@ -0,0 +1,2112 @@
+    return fillOutputString<CharType>(
+        in_value,
+        out_value,
+        out_value_max_length,
+        out_value_length,
+        length_in_bytes,
+        length_in_bytes,
+        true
+    );
+}
+
+// ObjectType, that is a pointer type, is treated as an integer, the value of that pointer.
+template <typename ObjectType, typename LengthType1, typename LengthType2>
+inline SQLRETURN fillOutputPOD(
+    const ObjectType & obj,
+    void * out_value,
+    LengthType1 out_value_max_length,
+    LengthType2 * out_value_length
+) {
+    return fillOutputBuffer(
+        &obj,
+        sizeof(obj),
+        out_value,
+        out_value_max_length,
+        out_value_length
+    );
+}
+
+template <typename ObjectType, typename LengthType1>
+inline SQLRETURN fillOutputPOD(
+    const ObjectType & obj,
+    void * out_value,
+    LengthType1 * out_value_length
+) {
+    return fillOutputPOD(
+        obj,
+        out_value,
+        sizeof(obj),
+        out_value_length
+    );
+}
+
+template <typename LengthType1, typename LengthType2>
+inline SQLRETURN fillOutputNULL(
+    void * out_value,
+    LengthType1 out_value_max_length,
+    LengthType2 * out_value_length
+) {
+    if (!out_value_length)
+        throw SqlException("Indicator variable required but not supplied", "22002");
+
+    *out_value_length = SQL_NULL_DATA;
+
+    return SQL_SUCCESS;
+}
+
+namespace value_manip {
+
+    template <typename T>
+    inline void to_null(T & obj) {
+        obj = T{};
+    }
+
+    inline void to_null(std::string & str) {
+        str.clear();
+    }
+
+    inline void to_null(SQL_NUMERIC_STRUCT & numeric) {
+        numeric.precision = 0;
+        numeric.scale = 0;
+        numeric.sign = 0;
+        std::fill(std::begin(numeric.val), std::end(numeric.val), 0);
+    }
+
+    inline void to_null(SQLGUID & guid) {
+        guid.Data1 = 0;
+        guid.Data2 = 0;
+        guid.Data3 = 0;
+        std::fill(std::begin(guid.Data4), std::end(guid.Data4), 0);
+    }
+
+    inline void to_null(SQL_DATE_STRUCT & date) {
+        date.year = 0;
+        date.month = 0;
+        date.day = 0;
+    }
+
+    inline void to_null(SQL_TIME_STRUCT & time) {
+        time.hour = 0;
+        time.minute = 0;
+        time.second = 0;
+    }
+
+    inline void to_null(SQL_TIMESTAMP_STRUCT & timestamp) {
+        timestamp.year = 0;
+        timestamp.month = 0;
+        timestamp.day = 0;
+        timestamp.hour = 0;
+        timestamp.minute = 0;
+        timestamp.second = 0;
+        timestamp.fraction = 0;
+    }
+
+    template <typename T>
+    inline void to_default(T & obj) {
+        return to_null(obj);
+    }
+
+    template <typename T>
+    static void normalize_date(T & date) {
+        if (date.year == 0)
+            date.year = 1970;
+
+        if (date.month == 0)
+            date.month = 1;
+
+        if (date.day == 0)
+            date.day = 1;
+    }
+
+} // namespace value_manip
+
+template <typename T>
+struct SimpleTypeWrapper {
```

Comment:
Is there any specific reason you need this wrapper? Why not just a type alias?

Reply:
To define `to_null()`'ing c-tor, and reuse code in general. Later, it is used as base for the most of `DataSourceType<>`, which in their turn used for better control over types and their static dispatch/overloads.